### PR TITLE
Secor testing

### PR DESCRIPTION
**Pull request by achad4**

### CODE REVIEW

File: src/main/java/com/pinterest/secor/util/ProtobufUtil.java [hunk 1]
```diff
@@ -130,22 +133,40 @@ public boolean isConfigured() {
...
+        } catch (IOException e) {
+            throw new RuntimeException("Error creating read stream for JSON message", e);
+        }
+    }
+
+    public Message decodeProtobufMessage(String topic, byte[] payload){
+        Method parseMethod = allTopics ? messageParseMethodForAll : messageParseMethodByTopic.get(topic);
+        try {
+            return (Message) parseMethod.invoke(null, payload);
+        } catch (IllegalArgumentException e) {
```
Comment (JohnnyMarnell) [inline src/main/java/com/pinterest/secor/util/ProtobufUtil.java]: I would collapse these with `|` and have same error message, error cases are close enough

File: src/main/java/com/pinterest/secor/util/ProtobufUtil.java [hunk 2]
```diff
@@ -130,22 +133,40 @@ public boolean isConfigured() {
...
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException("Can't parse protobuf message, since parseMethod() is not accessible. "
+                    + "Please check your protobuf version (this code works with protobuf >= 2.6.1)", e);
+        } catch (InvocationTargetException e) {
             throw new RuntimeException("Error parsing protobuf message", e);
         }
     }
 
     /**
      * Decodes protobuf message
```
Comment (JohnnyMarnell) [inline src/main/java/com/pinterest/secor/util/ProtobufUtil.java]: or JSON ?

File: src/main/java/com/pinterest/secor/util/ProtobufUtil.java (hunk 1 above)
Comment (achad4) [reply src/main/java/com/pinterest/secor/util/ProtobufUtil.java]: I agree, but this is how it is in pinterest's code, so would like to keep changes minimal for now

File: src/main/java/com/pinterest/secor/util/ProtobufUtil.java (hunk 2 above)
Comment (achad4) [reply src/main/java/com/pinterest/secor/util/ProtobufUtil.java]: what do you mean?

**Pull request merged**